lib/ts/recipe/session/with-jwt/recipeImplementation.ts

```diff
@@ -144,25 +158,23 @@ export default function (
             sessionHandle: string;
             newAccessTokenPayload: any;
         }): Promise<void> {
-            let sessionInformation = await this.getSessionInformation({ sessionHandle });
-
-            let jwtPropertyName = config.jwt.propertyNameInAccessTokenPayload;
-
-            if (config.jwt.getPropertyNameFromAccessTokenPayload !== undefined) {
-                jwtPropertyName = config.jwt.getPropertyNameFromAccessTokenPayload(
-                    sessionInformation.accessTokenPayload
-                );
+            if (newAccessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY] !== undefined) {
+                throw new Error(JWT_RESERVED_KEY_USE_ERROR_MESSAGE);
             }
 
-            let existingJWT = sessionInformation.accessTokenPayload[jwtPropertyName];
+            let sessionInformation = await this.getSessionInformation({ sessionHandle });
+
+            let jwtPropertyName = sessionInformation.accessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY];
 
-            if (existingJWT === undefined) {
+            if (jwtPropertyName === undefined || sessionInformation.accessTokenPayload[jwtPropertyName] === undefined) {
```

Comment:
You only need `if (jwtPropertyName === undefined)` and then put `sessionInformation.accessTokenPayload[jwtPropertyName] !== undefined` in an assert inside the `if` statement.

Reply:
The logic has been refactored, the assertion is added if the jwtPropertyName is not undefined